pkgs/by-name/gl/globalprotect-openconnect_2/package.nix

```diff
@@ -0,0 +1,122 @@
+{
+  perl,
+  jq,
+  fetchFromGitHub,
+  lib,
+  openconnect,
+  libsoup,
+  webkitgtk_4_0,
+  pkg-config,
+  callPackage,
+  rustPlatform,
+  glib,
+  atk,
+  gdk-pixbuf,
+  pango,
+  cairo,
+  harfbuzz,
+  gtk3,
+  zlib,
+  vpnc-scripts,
+  withGui ? false,
+}:
+let
+  platforms = [
+    "aarch64-linux"
+    "x86_64-linux"
+  ];
+  maintainers = with lib.maintainers; [
+    m1dugh
+    binary-eater
+  ];
+  version = "2.3.9";
+  pname = "globalprotect-openconnect";
+
+  gpgui = callPackage ./gui.nix { inherit version platforms maintainers; };
+
+in
+rustPlatform.buildRustPackage {
+  inherit version pname;
+
+  src = fetchFromGitHub {
+    owner = "yuezk";
+    repo = "GlobalProtect-openconnect";
+    rev = "v${version}";
+    hash = "sha256-s+uCpNrwQvdIINLSIbtcCCBg469J2xtlyiwDYqtXrQs=";
+  };
+
+  nativeBuildInputs = [
+    perl
+    jq
+    pkg-config
+  ];
+
+  buildInputs = [
+    libsoup.dev
+    webkitgtk_4_0
+    glib.dev
+    webkitgtk_4_0.dev
+    atk.dev
+    gdk-pixbuf.dev
+    pango.dev
+    cairo.dev
+    harfbuzz.dev
+    gtk3.dev
+    openconnect
+    zlib
+  ];
+
+  NIX_CFLAGS_COMPILE = (map (pkg: "-I${pkg}/include") [ openconnect.dev ]);
+
+  NIX_CFLAGS_LINK = (
+    map (pkg: "-L${lib.getLib pkg}/lib") [
+      openconnect
+      zlib
+    ]
+  );
+
+  cargoHash = "sha256-/KJM1JjFFI4aR1YwWI7k27QSPknPponTTazk5FG9HKc=";
+
+  postPatch =
+    let
+      replacements = [
+        "--replace-fail /usr/bin/gpclient $out/bin/gpclient"
+        "--replace-fail /usr/bin/gpservice $out/bin/gpservice"
+        "--replace-fail /usr/bin/gpgui-helper $out/bin/gpgui-helper"
+        "--replace-fail /usr/bin/gpauth $out/bin/gpauth"
+      ] ++ lib.optional withGui "--replace-fail /usr/bin/gpgui ${gpgui}/bin/gpgui";
+    in
+    ''
+      substituteInPlace crates/common/src/vpn_utils.rs \
+        --replace-fail /etc/vpnc/vpnc-script ${vpnc-scripts}/bin/vpnc-script
+
+      substituteInPlace crates/gpapi/src/lib.rs \
+        ${lib.strings.concatStringsSep " " replacements}
+    '';
+
+  postInstall =
+    ''
+      mkdir -p $out/share/applications/
+      cp packaging/files/usr/share/applications/gpgui.desktop $out/share/applications/gpgui.desktop
```

Comment:
https://github.com/NixOS/nixpkgs/blob/76612b17c0ce71689921ca12d9ffdc9c23ce40b2/pkgs/by-name/gp/gpclient/package.nix#L48-L51

You will need something like this `postFixup` to make this work.

Reply:
This is discouraged, per https://github.com/NixOS/nixpkgs/issues/308324. I suggest something like this:

```nix
   postFixup = '' 
     substituteInPlace $out/share/applications/gpgui.desktop \ 
       --replace-fail /usr/bin/gpclient gpclient 
   ''; 
```

Reply:
Thanks for sharing #308324. I agree with the notions there.